test/suite/test_live_restore09.py

```diff
@@ -0,0 +1,78 @@
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import os, time, wttest, threading
+from helper import copy_wiredtiger_home
+from wtbackup import backup_base
+
+# test_live_restore09.py
+# Test that crashing during the log copy phase won't break the logging subsystem.
+@wttest.skip_for_hook("tiered", "using multiple WT homes")
+class test_live_restore09(backup_base):
+    conn_config = 'log=(enabled=true)'
+
+    def test_live_restore09(self):
+        # Live restore is not supported on Windows.
+        if os.name == 'nt':
+            return
+        uri = "file:live_restore"
+        self.session.create(uri, 'key_format=i,value_format=S')
+        cursor = self.session.open_cursor(uri)
+        # Insert around a lot of data to leave us with several very big log files.
+        for i in range (0, 700000):
+            cursor[i] = "a" * 1000
+
+        # Close the default connection and back it up.
+        os.mkdir("SOURCE")
+        self.take_full_backup("SOURCE")
+        self.session.breakpoint()
+        self.close_conn()
+
+        # Make a directory to restore to.
+        os.mkdir("DEST")
+
+        # Start a thread to perform the restore, we need to do this in a separate thread at the
+        # call to open_conn will return after the temporary log files have been renamed. We want to
+        # catch the copy, prior to rename.
+        thread = threading.Thread(target=self.restore)
+        thread.start()
+        time.sleep(0.5)
+        # Copy over the WiredTiger home simulating a crash. We can't call simulate_crash_restart
+        # here as we need the thread to finish it's open prior to restarting.
+        copy_wiredtiger_home(self, "DEST", "RESTART")
+        # Wait for the open to finish.
+        thread.join()
+        # Close the newly opened connection and startup on then restart directory. This used to
+        # throw an error while trying to extract the log number from a temporary file.
+        self.close_conn()
+        self.open_conn("RESTART", config="statistics=(all),live_restore=(enabled=true,path=\"SOURCE\",threads_max=1)")
+        self.ignoreStdoutPatternIfExists('recreating metadata from backup')
```

Comment:
So interestingly "crashing" during startup leaves us in a state with both a turtle file and a backup file which apparently triggers some rarer, but safe, logic in `__wt_turtle_init` which logs a message to stdout in our python testing.

Reply:
It would recreate it from the metadata file which is recreated from the backup file, we probably do lose live restore state here but I think we're guaranteed to be in the setup state of live restore?

```    if (load || load_turtle) {
        /* Create the turtle file. */
        WT_ERR(__metadata_config(session, &metaconf));
        if (F_ISSET(conn, WT_CONN_LIVE_RESTORE_FS))
            ret = __wt_live_restore_turtle_update(session, WT_METAFILE_URI, metaconf, true);
        else
            WT_WITH_TURTLE_LOCK(
              session, ret = __wt_turtle_update(session, WT_METAFILE_URI, metaconf));
        __wt_free(session, metaconf);
        WT_ERR(ret);
```